examples/python/local_search.py

```
@@ -0,0 +1,46 @@
+    WebSearchOptions,
+)
+import os
+
+def local_search(query: str):
+    results = []
+    for root, _, files in os.walk('.'):
+        for f in files:
+            if query in f:
+                path = os.path.join(root, f)
+                try:
+                    content = open(path).read()
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)